### PR TITLE
Produce an event in case the swap yields a zero egress amount

### PR DESCRIPTION
**Pull request by RGafiyatullin**
# Pull Request

Closes: PRO-1075

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

### Existing functionality

If the deposited amount is less than `ingress-egress::MinimumDeposit` — an `ingress-egress::DepositIgnored { reason : BelowMinimumDeposit }` will be emitted.

If the (net) swapped amount is less than `swapping::MinimumSwapAmount` — a `swapping::SwapAmountTooLow` will be emitted (apparently, a deprecated scenario: PRO-1101).

### What this PR brings

If for any other reason, the `egress_amount` of a swap appears to be zero — a `swapping::EgressAmountZero` will be emitted.

### CODE REVIEW

**Comment by kylezs** [comment]
Tagging @chainflip-io/web here too - to confirm the swap id is all the info they need

**Review by kylezs — APPROVED** [review]

**Comment by kylezs** [comment]
(Await approval from a member of product team before merging)

**Review by niklasnatter — APPROVED** [review]
`swapId` should be enough for us to identify the swap in this case 👍

**Pull request merged**
